dotcom-rendering/src/web/components/PinnedPostLiveness.importable.tsx

```diff
@@ -38,5 +63,45 @@ export const PinnedPostLiveness = ({}) => {
 			);
 		};
 	}, []);
+
+	// calculate duration when user is viewing pinned post
+	// and emit ophan events when the pinned post goes out of view
+	useEffect(() => {
+		if (!pinnedPost) return () => {};
+
+		const pinnedPostTiming = initPerf('pinned-post-view-duration');
+
+		const observer = new IntersectionObserver(
+			([entry]) => {
+				if (entry.isIntersecting) {
+					hasBeenSeen.current = true;
+					pinnedPostTiming.clear();
+					pinnedPostTiming.start();
+				} else if (hasBeenSeen.current) {
+					const timeTaken = pinnedPostTiming.end();
+					if (timeTaken) {
+						const timeTakenInSeconds = timeTaken / 1000;
+						submitComponentEvent({
+							component: {
+								componentType: 'LIVE_BLOG_PINNED_POST',
+								id: pinnedPost?.id,
```

Comment:
I think you’re checking that `pinnedPost` isn’t undefined above

```suggestion
								id: pinnedPost.id,
```